### PR TITLE
feat(sdk): add clone_recurring_run method

### PR DESCRIPTION
**Pull request by droctothorpe**
Co-authored-by: js0823 <js0823@gmail.com>
Co-authored-by: travischeung <travis.cheung98@gmail.com>

**Description of your changes:**
There's no way to patch an existing recurring run via the SDK (or GUI), which is a little bit odd given that Argo Workflows supports patching CronWorkflows, but perhaps the goal is to make recurring runs immutable, which makes sense. The GUI gets around this with the `clone recurring run` button, but there's no equivalent in the SDK.

To address this feature gap, this PR adds a `clone_recurring_run` method to the SDK. The docstring is copied almost verbatim from the `create_recurring_run` method. 

We're completely amenable to feedback / input. If exposing patching capabilities makes more sense, we're happy to contribute to that as well.

We're also happy to add a `clone_run` method. 

On a separate note, bandwidth permitting, we might be able to start contributing unit test coverage to the SDK.

We'd also be happy to make a pass through of the SDK to contribute some improved formatting and line-wrapping to the docstrings. Happy to open a standalone issue if that seems reasonable. 

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by droctothorpe** [comment]
Pulled in changes from master. The co-authors will sign the CLA ASAP.

File: sdk/python/kfp/deprecated/_client.py
Comment (connor-mccarthy) [inline sdk/python/kfp/deprecated/_client.py:1242]: Consider using the following assignment expression throughout where possible for simplicity:
```python
job_name = job_name or f"Clone of {job.name}"
```

File: sdk/python/kfp/deprecated/_client.py
Comment (connor-mccarthy) [inline sdk/python/kfp/deprecated/_client.py:1322]: Consider using a context manager for this, in case an exception is thrown between file creation and cleanup.

**Review by connor-mccarthy — CHANGES_REQUESTED** [review]

**Comment by connor-mccarthy** [comment]
Thanks for your contribution, @droctothorpe! Just provided a few comments. I would like to have one more reviewer, if possible. Thank you again!

File: sdk/python/kfp/deprecated/_client.py
Comment (droctothorpe) [reply sdk/python/kfp/deprecated/_client.py]: Thanks for the spot on feedback, @connor-mccarthy! All comments addressed. Marking conversations as resolved.

**Comment by droctothorpe** [comment]
Just in case they got lost in the resolved conversations section, all comments were addressed. Side note: a very common use case for this capability is updating the cron string of an existing recurring run.

**Comment by ji-yaqi** [comment]
Hi @droctothorpe, for this interface change, could you add a feature request in issues and describe why this is necessary and the current workarounds? We want to make sure this interface change is well-documented before we merge in. (And we might need test coverages too). Thanks!

**Comment by droctothorpe** [comment]
> Hi @droctothorpe, for this interface change, could you add a feature request in issues and describe why this is necessary and the current workarounds? We want to make sure this interface change is well-documented before we merge in. (And we might need test coverages too). Thanks!

Makes perfect sense, @ji-yaqi . Sorry for jumping the gun! We have an opinionated subclass of Client and add features for our own benefit before submitting them upstream, so the work is already done, which is why we jumped straight to PR phase; we'll make sure to file issues first going forward. 

Here's the relevant issue:
https://github.com/kubeflow/pipelines/issues/7441

**Comment by droctothorpe** [comment]
Should we update this PR to target the [updated client](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/client/client.py)?

**Pull request closed without merging**
